src/lib/Backpack/devBackpack.cpp

```diff
@@ -90,12 +92,54 @@ void startPassthrough()
 }
 #endif
 
+#if defined(GPIO_PIN_BACKPACK_EN)
+
+static int debouncedRead(int pin) {
+    static const uint8_t sample_period_cycles = 100;
+    static const uint8_t min_matches = 2;
+
+    static uint8_t cycle = 0;
+    static int last_state = -1;
+    static uint8_t matches = 0;
+
+    int current_state;
+
+    if (cycle % sample_period_cycles == 0) {
+        // Skip reading the pin in this cycle.
+        cycle++;
+        return -1;
+    }
+
+    // Next read is only in sample_period_cycles.
+    cycle = 0;
+
+    current_state = digitalRead(pin);
+    if (current_state == last_state) {
+        matches = min(min_matches, (uint8_t)(matches + 1));
+    } else {
+        // We are bouncing. Reset the match counter.
+        matches = 0;
+    }
+
+    if (matches == min_matches) {
+        // We have a stable state and report it.
+        return current_state;
+    }
+    DBGLN("Bouncing!, current state: %d, last_state: %d, matches: %d", current_state, last_state, matches);
+
+    last_state = current_state;
+
+    // We don't have a definitive state we could report.
+    return -1;
+}
```

Comment:
Whilst this works, I think it's a little over-complicated.
All we really need is to read the pin, if it's high set the counter to 0 and return 0.
If it's low, increment the counter and if the counter is > 100 return -1 else return 0.

Reply:
Yeah, you are absolutely right. I would still cap the matches at min_matches to prevent an overflow from wrapping and resetting the counter to 0, and I would still treat high and low the same, so consider each state only stable when 100 cycles have the same value. Updated the PR accordingly, let me know what you think